sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileStatusCache.scala

```diff
@@ -94,13 +94,25 @@ private class SharedInMemoryCache(maxSizeInBytes: Long) extends Logging {
   // Opaque object that uniquely identifies a shared cache user
   private type ClientId = Object
 
+  /* [[Weigher]].weigh returns Int so we could only cache objects < 2GB
+   * instead, the weight is divided by this factor (which is smaller
+   * than the size of one [[FileStatus]]).
+   * so it will support objects up to 64GB in size.
+   */
+  private val weightScale = 32
```

Comment:
Rather than make it a member variable, just a local variable in the initializer for cache?